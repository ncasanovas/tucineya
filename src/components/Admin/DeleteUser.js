import React, { useState } from "react";
import axios from "axios";

import { Modal, Button } from "react-bootstrap";
import { useHistory } from "react-router";

export const DeleteUser = ({ deleteUsers, setDeleteUsers }) => {
  const [show, setShow] = useState(false);

  const history = useHistory();

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const handleDelete = () => {
    if (deleteUsers.length !== 0) {
      deleteUsers.map(async (user) => {
        await axios
          .delete(
            `https://tucineya.herokuapp.com/api/users/${user}`
            //`http://localhost:4000/api/users/${user}`
          )
          .catch((e) => {
            console.log(e);
          });
      });
    }

    setDeleteUsers([]);
    handleClose();
    history.go(0);
  };

  return (
    <div className="mt-2">
      <Button
        variant={deleteUsers.length === 0 ? "primary disabled" : "primary"}
        onClick={handleShow}
      >
        Eliminar Usuario/s
      </Button>

      <Modal
        show={show}
        onHide={handleClose}
        backdrop="static"
        keyboard={false}
      >
        <Modal.Header closeButton>
          <Modal.Title>¡Atencion!</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          ¿Está seguro de querer eliminar los usuarios seleccionados?
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary btn-secondary" onClick={handleClose}>
            Volver
          </Button>
          <Button variant="danger" onClick={handleDelete}>
            Eliminar
          </Button>
        </Modal.Footer>
      </Modal>
    </div>
  );
};
