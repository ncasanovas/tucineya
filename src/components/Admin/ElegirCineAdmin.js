import React, { useState, useEffect } from "react";
import { Dropdown } from "react-bootstrap";
import axios from "axios";

export const ElegirCineAdmin = ({ setIdCine }) => {
  const [localidades, setLocalidades] = useState();
  const [barrio, setBarrio] = useState([]);

  useEffect(async () => {
    await axios
      //.get("http://localhost:4000/api/cines/")
      .get("https://tucineya.herokuapp.com/api/cines/")
      .then((res) => {
        setLocalidades(res.data[0]);
      });
  }, []);

  const onClick = async (localidad) => {
    await axios
      //.get(`http://localhost:4000/api/cines/${localidad.idLocalidad}`)
      .get(`https://tucineya.herokuapp.com/api/cines/${localidad.idLocalidad}`)
      .then((res) => {
        setBarrio(res.data[0]);
      });
  };

  const barrios = (barrio) => {
    setIdCine(barrio.idCine);
  };

  return (
    <Dropdown>
      <Dropdown.Toggle variant="primary" id="dropdown-basic">
        {barrio.length === 0 ? "Localidades" : barrio[0].Localidad}
      </Dropdown.Toggle>

      <Dropdown.Menu>
        {localidades === undefined
          ? null
          : localidades.map((localidad, i) => {
              return (
                <Dropdown.Item key={i} onClick={() => onClick(localidad)}>
                  {localidad.nombreLocalidad}
                </Dropdown.Item>
              );
            })}
      </Dropdown.Menu>
      <div>
        <ul className="list-group list-group-flush">
          {barrio.length === 0
            ? null
            : barrio.map((barrio, i) => {
                return (
                  <div key={i}>
                    <li
                      className="list-group-item"
                      onClick={() => barrios(barrio)}
                      style={{ cursor: "pointer" }}
                    >
                      {barrio.Barrio}
                      {" - "}
                      {barrio.nombre}
                    </li>
                  </div>
                );
              })}
        </ul>
      </div>
    </Dropdown>
  );
};
