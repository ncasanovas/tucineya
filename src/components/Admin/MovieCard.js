import React from "react";
import axios from "axios";
import { Card, Button } from "react-bootstrap";

export const MovieCard = ({ movie }) => {
  const onClickAddMovie = async () => {
    await axios
      .post("http://localhost:4000/api/movies/", movie)
      //.post("https://tucineya.herokuapp.com/api/movies/", movie)
      .then((data) => {
        alert(data.data.message);
      })
      .catch((e) => {
        console.log(e);
      });
  };

  return (
    <div className="mb-3">
      <div className="d-flex justify-content-center">
        <Card className="row g-0 ">
          <div className="col-md-4">
            <Card.Img
              src={`https://image.tmdb.org/t/p/w300${movie.poster_path}`}
            />
          </div>
          <div className="col-md-8">
            <Card.Body>
              <Card.Title>{movie.title}</Card.Title>
              <Card.Text>
                <b>Voto Promedio:</b> {movie.vote_average}
              </Card.Text>
              <Card.Text>
                <b>Fecha de estreno:</b> {movie.release_date}
              </Card.Text>
            </Card.Body>
            <Button variant="primary" onClick={onClickAddMovie}>
              Agregar Pelicula
            </Button>
          </div>
        </Card>
      </div>
    </div>
  );
};
