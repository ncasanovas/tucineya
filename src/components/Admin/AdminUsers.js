import React, { useState, useEffect } from "react";
import axios from "axios";

import { DeleteUser } from "./DeleteUser";

export const AdminUsers = () => {
  const [users, setUsers] = useState();
  const [deleteUsers, setDeleteUsers] = useState([]);

  useEffect(async () => {
    await axios
      .get("https://tucineya.herokuapp.com/api/users/")
      //.get("http://localhost:4000/api/users/")
      .then((res) => {
        setUsers(res.data[0]);
      });
  }, []);

  const onChangeCheck = (e) => {
    if (e.target.checked) {
      setDeleteUsers([...deleteUsers, e.target.value]);
    } else {
      for (let i = 0; i < deleteUsers.length; i++) {
        if (deleteUsers[i] === e.target.value) {
          deleteUsers.splice(i, 1);
          setDeleteUsers([...deleteUsers]);
        }
      }
    }
  };

  return (
    <div className="mb-3">
      <div>
        {users ? (
          <div className="overflow-auto d-flex container p-2">
            <ul className="list-group text-start">
              {users.map((user, i) => {
                return (
                  <li className="list-group-item" key={i}>
                    <input
                      className="form-check-input me-1"
                      type="checkbox"
                      value={user.mail}
                      aria-label="..."
                      onChange={onChangeCheck}
                    />
                    {user.nombre} {user.apellido}
                  </li>
                );
              })}
            </ul>
          </div>
        ) : (
          <div
            className="spinner-border text-primary d-flex justify-content-center"
            role="status"
          >
            <span className="visually-hidden ">Loading...</span>
          </div>
        )}
      </div>
      <div className="d-flex justify-content-center">
        {users ? (
          <DeleteUser
            deleteUsers={deleteUsers}
            setDeleteUsers={setDeleteUsers}
          />
        ) : null}
      </div>
    </div>
  );
};
