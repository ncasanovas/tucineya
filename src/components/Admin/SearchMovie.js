import React, { useState } from "react";
import axios from "axios";

import DatePicker from "react-date-picker";

import { Modal, Button } from "react-bootstrap";
import { MovieCard } from "./MovieCard";
import { ElegirCineAdmin } from "./ElegirCineAdmin";

export const SearchMovie = () => {
  const [show, setShow] = useState(false);
  const [inputValue, setInputValue] = useState();
  const [movieResult, setMovieResult] = useState();
  const [peli, setPeli] = useState();
  const [idCine, setIdCine] = useState();
  const [value, onChange] = useState(new Date());

  const handleClose = () => {
    setShow(false);
    setInputValue("");
    setMovieResult();
    setPeli();
    setIdCine();
  };
  const handleShow = () => setShow(true);

  const agregarPelicula = async () => {
    await axios
      /* .post("http://localhost:4000/api/movies/", {
        titulo: peli.title,
        sinopsis: peli.overview,
        poster: peli.poster_path,
        idPelicula: peli.id,
        idCine: idCine,
        fecha: value,
      }) */
      .post("https://tucineya.herokuapp.com/api/movies/", {
        titulo: peli.title,
        sinopsis: peli.overview,
        poster: peli.poster_path,
        idPelicula: peli.id,
        idCine: idCine,
        fecha: value,
      })
      .then((data) => {
        alert(data.data.message);
      })
      .catch((e) => {
        console.log(e);
      });
  };

  const onChangeinput = (e) => {
    setInputValue(e.target.value);
  };

  const onClickSearchMovie = async () => {
    await axios
      .get(
        `https://api.themoviedb.org/3/search/movie?api_key=350f655333437185ccf33d95346bf8e6&language=en-US&query=${inputValue}&page=1&include_adult=false`
      )
      .then((res) => {
        setMovieResult(res.data.results);
      })
      .catch((e) => {
        console.log(e);
      });
  };

  return (
    <div className="mt-2">
      <Button variant={"primary"} onClick={handleShow}>
        Agregar Pelicula
      </Button>

      <Modal
        show={show}
        onHide={handleClose}
        backdrop="static"
        keyboard={false}
      >
        <Modal.Header closeButton>
          <Modal.Title>Pelicula</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          {peli ? (
            <div className="row">
              <div className="col">
                <img
                  src={`https://image.tmdb.org/t/p/w300${peli.poster_path}`}
                  alt={peli.title}
                  className="imgPelicula"
                  onDoubleClick={() => setPeli()}
                />
              </div>
              <div className="col">
                <h4>Elige la fecha y el cine donde se estrena</h4>
                <DatePicker
                  onChange={onChange}
                  value={value}
                  className="mb-3"
                  format="y-MM-dd"
                />
                <ElegirCineAdmin setIdCine={setIdCine} />
              </div>
            </div>
          ) : (
            <div className="mb-3">
              <input
                type="text"
                placeholder="Busque por titulo"
                onChange={onChangeinput}
              />

              <Button
                variant="secondary btn-secondary"
                onClick={onClickSearchMovie}
              >
                Buscar
              </Button>
            </div>
          )}

          <div>
            {movieResult
              ? movieResult.map((movie, i) => {
                  return (
                    <MovieCard
                      key={i}
                      movie={movie}
                      setPeli={setPeli}
                      setMovieResult={setMovieResult}
                    />
                  );
                })
              : null}
          </div>
        </Modal.Body>
        <Modal.Footer>
          <Button
            variant="secondary btn-secondary"
            onClick={() => {
              agregarPelicula();
            }}
          >
            Agregar Pelicula
          </Button>
          <Button variant="secondary btn-secondary" onClick={handleClose}>
            Volver
          </Button>
        </Modal.Footer>
      </Modal>
    </div>
  );
};
