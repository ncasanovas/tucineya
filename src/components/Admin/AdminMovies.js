import React, { useState, useEffect } from "react";
import { SearchMovie } from "./SearchMovie";
import axios from "axios";

export const AdminMovies = () => {
  const [movies, setMovies] = useState();

  useEffect(async () => {
    await axios
      .get("https://tucineya.herokuapp.com/api/movies/")
      //.get("http://localhost:4000/api/movies/")
      .then((res) => {
        setMovies(res.data[0]);
      });
  });

  const onClickDeleteMovie = async (idNombrePelicula) => {
    await axios
      //.delete(`http://localhost:4000/api/movies/${idNombrePelicula}`)
      .delete(`https://tucineya.herokuapp.com/api/movies/${idNombrePelicula}`)
      .then((res) => {
        alert(res.data.message);
      })
      .catch((e) => {
        console.log(e);
      });
  };

  return (
    <div>
      <div>
        {movies ? (
          <div
            id="withContext"
            className="overflow-auto d-flex container p-2"
            style={{ height: "350px", width: "auto" }}
          >
            <div className="row row-cols-3 row-cols-lg-5 row-cols-md-4 g-2 g-lg-3">
              {movies.map((movie, i) => {
                return (
                  <div key={i}>
                    <div>
                      <img
                        src={movie.posterPelicula}
                        className="card-img-top"
                        alt={movie.idNombrePelicula}
                      />
                      <div className="card-body p-0 ">
                        <div className="d-flex justify-content-center  m-2">
                          <h6 className="card-title">
                            {movie.idNombrePelicula}
                          </h6>
                        </div>
                        <div>
                          <button
                            className="btn btn-danger "
                            onClick={() =>
                              onClickDeleteMovie(movie.idNombrePelicula)
                            }
                          >
                            Eliminar
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                );
              })}
            </div>
          </div>
        ) : (
          <div className="d-flex justify-content-center">
            <div
              className="spinner-border text-primary d-flex justify-content-center"
              role="status"
            >
              <span className="visually-hidden ">Loading...</span>
            </div>
          </div>
        )}
      </div>
      {movies ? <SearchMovie /> : null}
    </div>
  );
};
