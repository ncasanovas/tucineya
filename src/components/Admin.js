import React, { useState, useEffect, useContext } from "react";
import axios from "axios";
import { HashRouter, Link } from "react-router-dom";

import { AdminModal } from "./AdminModal";
import { AdminMovies } from "./AdminMovies";
import { MoviesContext } from "./MoviesContext";

export const Admin = () => {
  const [users, setUsers] = useState();
  const [deleteUsers, setDeleteUsers] = useState([]);

  const { setMoviesContext } = useContext(MoviesContext);

  useEffect(() => {
    const data = async () => {
      await axios
        //.get("https://tucineya.herokuapp.com/api/users/")
        .get("http://localhost:4000/api/users/")
        .then((res) => {
          setUsers(res.data[0]);
        });
    };
    const datamovies = async () => {
      await axios
        //.get("https://tucineya.herokuapp.com/api/movies/")
        .get("http://localhost:4000/api/movies/")
        .then((res) => {
          setMoviesContext(res.data[0]);
        });
    };
    data();
    datamovies();
  }, []);

  const onChangeCheck = (e) => {
    if (e.target.checked) {
      setDeleteUsers([...deleteUsers, e.target.value]);
    } else {
      for (let i = 0; i < deleteUsers.length; i++) {
        if (deleteUsers[i] === e.target.value) {
          deleteUsers.splice(i, 1);
          setDeleteUsers([...deleteUsers]);
        }
      }
    }
  };

  return (
    <div id="admin" className="container-fluid row">
      <div className="d-flex justify-content-end">
        <HashRouter>
          <Link to="/" className="col-2">
            Atras
          </Link>
        </HashRouter>
      </div>
      <div className="col-3 col-sm-4 col-md-3 col-lg-3">
        <h4 className="mb-4">Lista de usuarios</h4>
        <div className="row">
          {users ? (
            <ul className="list-group text-start col">
              {users.map((user, i) => {
                return (
                  <li className="list-group-item" key={i}>
                    <input
                      className="form-check-input me-1"
                      type="checkbox"
                      value={user.mail}
                      aria-label="..."
                      onChange={onChangeCheck}
                    />
                    {user.nombre} {user.apellido}
                  </li>
                );
              })}
            </ul>
          ) : (
            <div className="spinner-border text-primary m-6" role="status">
              <span className="visually-hidden ">Loading...</span>
            </div>
          )}
        </div>
        <div>
          {users ? (
            <div>
              <AdminModal
                deleteUsers={deleteUsers}
                setDeleteUsers={setDeleteUsers}
              />
            </div>
          ) : null}
        </div>
      </div>
      <div className="col-sm-12 col-md-9 col-lg-9">
        <h4 className="mb-4">Peliculas</h4>
        <AdminMovies />
      </div>
    </div>
  );
};

export default Admin;
