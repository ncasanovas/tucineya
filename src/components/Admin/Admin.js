import React, { useState, useEffect, useContext } from "react";
import axios from "axios";
import { HashRouter, Link } from "react-router-dom";

import { AdminMovies } from "./AdminMovies";
import { MoviesContext } from "../MoviesContext";
import { AdminUsers } from "./AdminUsers";

export const Admin = () => {
  const [users, setUsers] = useState();

  const { moviesContext, setMoviesContext } = useContext(MoviesContext);

  useEffect(() => {
    const data = async () => {
      await axios
        //.get("https://tucineya.herokuapp.com/api/users/")
        .get("http://localhost:4000/api/users/")
        .then((res) => {
          setUsers(res.data[0]);
        });
    };
    const datamovies = async () => {
      await axios
        //.get("https://tucineya.herokuapp.com/api/movies/")
        .get("http://localhost:4000/api/movies/")
        .then((res) => {
          setMoviesContext(res.data[0]);
        });
    };
    data();
    datamovies();
  }, [moviesContext]);

  return (
    <div className="container-fluid row">
      <div className="d-flex justify-content-end">
        <HashRouter>
          <Link to="/" className="col-2">
            Atras
          </Link>
        </HashRouter>
      </div>
      <div className="col-sm-7 col-md-3 col-lg-3">
        <h4 className="mb-4">Lista de usuarios</h4>
        <AdminUsers users={users} />
      </div>
      <div className="col-sm-12 col-md-9 col-lg-9">
        <h4 className="mb-4">Peliculas</h4>
        <AdminMovies />
      </div>
    </div>
  );
};

export default Admin;
