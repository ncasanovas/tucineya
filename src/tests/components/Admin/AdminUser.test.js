import React from "react";

import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import axios from "axios";

import { AdminUsers } from "../../../components/Admin/AdminUsers";

describe("Pruebas en <AdminUser/>", () => {
  const userTest = [
    {
      nombre: "test",
      apellido: "test",
      mail: "test@test.com",
      username: "test",
    },
  ];
  const wrapper = shallow(<AdminUsers user={userTest} />);

  test("Debe mostrarse correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar los usuarios de la base de datos ", async () => {
    jest.setTimeout(30000);
    await axios.get("https://tucineya.herokuapp.com/api/users/").then((res) => {
      expect(res.data).not.toEqual([[]]);
      expect(typeof res.data.length).toBe("number");
    });
  });
});
