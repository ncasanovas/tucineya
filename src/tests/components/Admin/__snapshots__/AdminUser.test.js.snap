// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AdminUser/> Debe mostrarse correctamente el componente 1`] = `
<div
  className="mb-3"
>
  <div>
    <div
      className="spinner-border text-primary d-flex justify-content-center"
      role="status"
    >
      <span
        className="visually-hidden "
      >
        Loading...
      </span>
    </div>
  </div>
  <div
    className="d-flex justify-content-center"
  />
</div>
`;
